src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.ts

```diff
@@ -102,22 +115,50 @@ export class BaseAdapter<
 
     const pair = this.store.getState().advancedTradingView.adapters[this._key][pairId]
 
-    if ((pair && !isFirstFetch && !pair.burnsAndMints?.hasMore) || (pair && isFirstFetch)) return
-
     try {
-      const burnsAndMints = await this._fetchBurnsAndMints({
-        pairId,
-        pair,
-        chainId: this._chainId,
-        amountToFetch,
-        abortController,
-        inputTokenAddress: inputToken.address,
-        outputTokenAddress: outputToken.address,
-      })
-
-      this._dispatchBurnsAndMints(pairId, burnsAndMints, amountToFetch)
+      if (payloadType === AdapterPayloadType.SWAPS) {
+        const { swaps } = await this._fetchSwaps({
+          pairId,
+          pair,
+          chainId: this._chainId,
+          amountToFetch,
+          abortController,
+          refreshing,
+          inputTokenAddress: inputToken.address,
+          outputTokenAddress: outputToken.address,
+        })
+
+        return {
+          payloadType: AdapterPayloadType.SWAPS,
+          key: this._key,
+          pairId,
+          data: swaps,
+          hasMore: swaps.length === amountToFetch,
```

Comment:
does this tell us if it has more to fetch?

Reply:
Yeah, it is used when you scroll and look for the past data.